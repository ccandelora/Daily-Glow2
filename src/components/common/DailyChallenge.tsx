import React, { useState, useEffect } from 'react';
import { View, StyleSheet, TouchableOpacity, Animated, ViewStyle, StyleProp } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Typography } from './Typography';
import { Card } from './Card';
import { Button } from './Button';
import { Input } from './Input';
import { useChallenges } from '@/contexts/ChallengesContext';
import { useAppState } from '@/contexts/AppStateContext';
import theme from '@/constants/theme';
import { Ionicons } from '@expo/vector-icons';
import { useRouter } from 'expo-router';

interface DailyChallengeProps {
  onComplete?: () => void;
  style?: StyleProp<ViewStyle>;
}

const typeIcons = {
  mood: 'heart-outline',
  gratitude: 'happy-outline',
  mindfulness: 'leaf-outline',
  creative: 'brush-outline',
} as const;

const getPromptForType = (type: string) => {
  switch (type) {
    case 'mood':
      return 'Describe your current mood and feelings...';
    case 'gratitude':
      return 'Share what you\'re grateful for today...';
    case 'mindfulness':
      return 'Describe your mindfulness experience...';
    case 'creative':
      return 'Share your creative response...';
    default:
      return 'Share your thoughts...';
  }
};

const getMinLengthForType = (type: string) => {
  return type === 'creative' ? 20 : 10;
};

export const DailyChallenge: React.FC<DailyChallengeProps> = ({ onComplete, style }) => {
  const { dailyChallenge, userStats, completeChallenge, refreshDailyChallenge, userChallenges } = useChallenges();
  const { showError } = useAppState();
  const [completionText, setCompletionText] = useState('');
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [localPoints, setLocalPoints] = useState(0);
  const [completedToday, setCompletedToday] = useState(0);
  const [showLimitMessage, setShowLimitMessage] = useState(false);
  const router = useRouter();

  useEffect(() => {
    console.log('DailyChallenge state:', {
      hasDailyChallenge: !!dailyChallenge,
      dailyChallengeId: dailyChallenge?.id,
      completedToday,
      showLimitMessage,
      userChallengesCount: userChallenges.length
    });
  }, []);

  useEffect(() => {
    if (userStats?.total_points !== undefined) {
      setLocalPoints(userStats.total_points);
      
      // Count completed challenges for today using local timezone
      const today = new Date();
      today.setHours(0, 0, 0, 0); // Set to start of day in local timezone
      
      const completedCount = userChallenges.filter(challenge => {
        if (!challenge.completed_at) return false;
        
        // Convert UTC timestamp to local date for comparison
        const completionDate = new Date(challenge.completed_at);
        const localCompletionDate = new Date(
          completionDate.getFullYear(),
          completionDate.getMonth(),
          completionDate.getDate()
        );
        
        const isCompleted = challenge.status === 'completed';
        const isToday = localCompletionDate.getTime() === today.getTime();
        
        return isCompleted && isToday;
      }).length;
      
      setCompletedToday(completedCount);
      setShowLimitMessage(completedCount >= 2);
    }
  }, [userStats, userChallenges]);

  // Single useEffect for refresh logic
  useEffect(() => {
    const setupMidnightRefresh = () => {
      const now = new Date();
      const tomorrow = new Date(now);
      tomorrow.setDate(tomorrow.getDate() + 1);
      tomorrow.setHours(0, 0, 0, 0);
      
      const timeUntilMidnight = tomorrow.getTime() - now.getTime();
      
      // Only refresh if we don't have a challenge
      if (!dailyChallenge) {
        refreshDailyChallenge();
      }
      
      return setTimeout(() => {
        refreshDailyChallenge();
        setupMidnightRefresh();
      }, timeUntilMidnight);
    };

    const timer = setupMidnightRefresh();
    return () => clearTimeout(timer);
  }, [dailyChallenge]); // Only re-run if dailyChallenge changes

  if (showLimitMessage) {
    return (
      <View style={styles.glowContainer}>
        <Card style={styles.container}>
          <View style={styles.completionContent}>
            <Ionicons 
              name="checkmark-circle-outline" 
              size={48} 
              color={theme.COLORS.primary.green}
              style={styles.completionIcon}
            />
            <Typography variant="h2" style={styles.completionTitle}>
              Daily Challenges Complete!
            </Typography>
            <Typography style={styles.completionMessage}>
              Great job! You've completed your daily challenges. Come back tomorrow for new challenges and more opportunities to earn points.
            </Typography>
            <View style={styles.completionPoints}>
              <Typography variant="h1" style={styles.completionPointsValue}>
                {localPoints}
              </Typography>
              <Typography style={styles.completionPointsLabel}>
                Total Points
              </Typography>
            </View>
            <Button
              title="View Your Achievements"
              onPress={() => router.push('/(app)/achievements')}
              variant="secondary"
              style={styles.completionButton}
            />
          </View>
        </Card>
      </View>
    );
  }

  if (!dailyChallenge) {
    return (
      <View style={styles.glowContainer}>
        <Card style={styles.container}>
          <Typography variant="body" style={styles.limitMessage}>
            No challenges available right now. Please check back later.
          </Typography>
        </Card>
      </View>
    );
  }

  const handleComplete = async () => {
    if (completedToday >= 2) {
      setShowLimitMessage(true);
      return;
    }

    try {
      setIsSubmitting(true);
      
      const minLength = getMinLengthForType(dailyChallenge.type);
      if (!completionText.trim() || completionText.length < minLength) {
        showError(`Please provide a response with at least ${minLength} characters`);
        return;
      }

      await completeChallenge(dailyChallenge.id, completionText.trim());
      await refreshDailyChallenge();
      setCompletionText('');
      onComplete?.();
    } catch (error) {
      if (error instanceof Error && error.message.includes('Daily challenge limit reached')) {
        setShowLimitMessage(true);
      } else {
        showError(error instanceof Error ? error.message : 'Failed to complete challenge');
      }
    } finally {
      setIsSubmitting(false);
    }
  };

  const currentStreak = userStats?.current_streak ?? 0;
  const level = userStats?.level ?? 1;

  return (
    <View style={styles.glowContainer}>
      <Card style={[styles.container, style]}>
        <View style={styles.header}>
          <View style={styles.titleContainer}>
            <Ionicons 
              name={typeIcons[dailyChallenge.type as keyof typeof typeIcons]} 
              size={24} 
              color={theme.COLORS.primary.green}
            />
            <Typography variant="h3" style={styles.title}>
              Daily Challenge
            </Typography>
          </View>
          <View style={styles.challengeStats}>
            <Typography variant="caption" color={theme.COLORS.ui.textSecondary}>
              Level {level}
            </Typography>
            <Typography 
              variant="h3" 
              color={theme.COLORS.primary.green}
              style={styles.pointsText}
            >
              {localPoints} pts
            </Typography>
          </View>
        </View>

        <Typography variant="h3" style={styles.challengeTitle}>
          {dailyChallenge.title}
        </Typography>
        
        <Typography 
          variant="body" 
          style={{ color: theme.COLORS.ui.textSecondary }}
        >
          {dailyChallenge.description}
        </Typography>

        <Input
          multiline
          value={completionText}
          onChangeText={setCompletionText}
          placeholder={getPromptForType(dailyChallenge.type)}
          style={styles.input}
          maxLength={500}
        />

        <View style={styles.footer}>
          <View style={styles.pointsContainer}>
            <Ionicons 
              name="star-outline" 
              size={16} 
              color={theme.COLORS.primary.green}
            />
            <Typography 
              variant="body"
              style={styles.points}
              color={theme.COLORS.primary.green}
            >
              {dailyChallenge.points} points
            </Typography>
          </View>
          <Button
            title="Complete Challenge"
            onPress={handleComplete}
            style={styles.button}
            disabled={isSubmitting}
          />
        </View>

        {currentStreak > 0 && (
          <View style={styles.streakContainer}>
            <Ionicons 
              name="flame-outline" 
              size={16} 
              color={theme.COLORS.primary.red}
            />
            <Typography 
              variant="caption" 
              style={{ color: theme.COLORS.primary.red }}
            >
              {currentStreak} day streak!
            </Typography>
          </View>
        )}
      </Card>
    </View>
  );
};

const styles = StyleSheet.create({
  glowContainer: {
    width: '100%',
    alignItems: 'center',
    marginVertical: theme.SPACING.lg,
    paddingHorizontal: theme.SPACING.lg,
  },
  container: {
    width: '100%',
    padding: theme.SPACING.xl,
    backgroundColor: 'rgba(255, 255, 255, 0.95)',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginBottom: theme.SPACING.md,
    width: '100%',
    flexWrap: 'wrap',
    gap: theme.SPACING.md,
  },
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: theme.SPACING.sm,
    flex: 1,
    minWidth: 150,
  },
  title: {
    marginLeft: theme.SPACING.xs,
    flexShrink: 1,
  },
  challengeStats: {
    alignItems: 'flex-end',
    minWidth: 100,
    flexShrink: 0,
  },
  challengeTitle: {
    marginBottom: theme.SPACING.sm,
  },
  description: {
    marginBottom: theme.SPACING.lg,
  },
  input: {
    height: 120,
    marginBottom: theme.SPACING.lg,
    textAlignVertical: 'top',
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  pointsContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    gap: theme.SPACING.xs,
  },
  points: {
    fontWeight: theme.FONTS.weights.semibold,
    fontSize: theme.FONTS.sizes.md,
  },
  button: {
    flex: 1,
    marginLeft: theme.SPACING.md,
  },
  streakContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: theme.SPACING.md,
    paddingTop: theme.SPACING.md,
    borderTopWidth: 1,
    borderTopColor: theme.COLORS.ui.border,
  },
  pointsText: {
    fontWeight: theme.FONTS.weights.bold,
    color: theme.COLORS.primary.green,
    marginTop: theme.SPACING.xs,
    textAlign: 'right',
    minWidth: 100,
    flexWrap: 'wrap',
    flexShrink: 0,
  },
  limitContainer: {
    padding: theme.SPACING.lg,
    alignItems: 'center',
    width: '100%',
  },
  limitHeader: {
    alignItems: 'center',
    marginBottom: theme.SPACING.xl,
    width: '100%',
  },
  limitTitle: {
    textAlign: 'center',
    color: theme.COLORS.primary.green,
    marginTop: theme.SPACING.md,
    fontSize: theme.FONTS.sizes.xxl,
    paddingHorizontal: theme.SPACING.md,
    width: '100%',
    flexWrap: 'wrap',
  },
  limitMessage: {
    textAlign: 'center',
    color: theme.COLORS.ui.textSecondary,
    marginBottom: theme.SPACING.xl,
    lineHeight: 24,
    paddingHorizontal: theme.SPACING.lg,
    width: '100%',
  },
  statsContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    marginBottom: theme.SPACING.xl,
    paddingHorizontal: theme.SPACING.lg,
  },
  pointsDisplay: {
    alignItems: 'center',
    flex: 1,
    paddingHorizontal: theme.SPACING.lg,
    width: '100%',
    maxWidth: 200,
  },
  pointsValue: {
    color: theme.COLORS.primary.green,
    fontSize: theme.FONTS.sizes.xxxl,
    fontWeight: theme.FONTS.weights.bold,
    textAlign: 'center',
    width: '100%',
    flexShrink: 0,
    minWidth: 150,
    paddingHorizontal: theme.SPACING.md,
  },
  pointsLabel: {
    color: theme.COLORS.ui.textSecondary,
    marginTop: theme.SPACING.xs,
    textAlign: 'center',
    width: '100%',
  },
  streakDisplay: {
    alignItems: 'center',
    flex: 1,
    borderLeftWidth: 1,
    borderLeftColor: theme.COLORS.ui.border,
    paddingLeft: theme.SPACING.lg,
  },
  streakValue: {
    color: theme.COLORS.primary.red,
    marginTop: theme.SPACING.xs,
    fontSize: theme.FONTS.sizes.xxl,
    fontWeight: theme.FONTS.weights.bold,
  },
  streakLabel: {
    color: theme.COLORS.ui.textSecondary,
  },
  achievementsButton: {
    marginTop: 0,
    paddingVertical: theme.SPACING.md,
    width: '100%',
  },
  completionContainer: {
    width: '100%',
    backgroundColor: 'rgba(255, 255, 255, 0.95)',
    borderRadius: theme.BORDER_RADIUS.lg,
  },
  completionContent: {
    padding: theme.SPACING.xl,
    alignItems: 'center',
    paddingVertical: theme.SPACING.xl * 1.5,
  },
  completionIcon: {
    marginBottom: theme.SPACING.lg,
  },
  completionTitle: {
    color: theme.COLORS.primary.green,
    fontSize: 28,
    fontWeight: theme.FONTS.weights.semibold,
    textAlign: 'center',
    marginBottom: theme.SPACING.md,
  },
  completionMessage: {
    color: theme.COLORS.ui.textSecondary,
    textAlign: 'center',
    marginBottom: theme.SPACING.xl * 1.5,
    lineHeight: 24,
    fontSize: theme.FONTS.sizes.md,
  },
  completionPoints: {
    alignItems: 'center',
    marginBottom: theme.SPACING.xl * 1.5,
  },
  completionPointsValue: {
    color: theme.COLORS.primary.green,
    fontSize: 72,
    fontWeight: theme.FONTS.weights.bold,
    textAlign: 'center',
    marginBottom: theme.SPACING.xs,
    lineHeight: 80,
  },
  completionPointsLabel: {
    color: theme.COLORS.ui.textSecondary,
    textAlign: 'center',
    fontSize: theme.FONTS.sizes.lg,
  },
  completionButton: {
    width: '100%',
    paddingVertical: theme.SPACING.md,
    marginTop: theme.SPACING.md,
  },
}); 