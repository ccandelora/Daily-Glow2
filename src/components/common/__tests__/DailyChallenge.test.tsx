import React from 'react';
import { render, fireEvent, act, waitFor } from '@testing-library/react-native';
import { DailyChallenge } from '../DailyChallenge';

// Mock dependencies
jest.mock('@/contexts/ChallengesContext', () => ({
  useChallenges: jest.fn(),
}));

jest.mock('@/contexts/AppStateContext', () => ({
  useAppState: jest.fn().mockReturnValue({
    showError: jest.fn(),
    showSuccess: jest.fn(),
  }),
}));

jest.mock('expo-router', () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
  }),
}));

// Mock FontAwesome6 icons
jest.mock('@expo/vector-icons', () => ({
  FontAwesome6: 'FontAwesome6Mock',
}));

// Mock LinearGradient
jest.mock('expo-linear-gradient', () => ({
  LinearGradient: 'LinearGradientMock',
}));

// Mock common components
jest.mock('../Typography', () => ({
  Typography: 'TypographyMock',
}));

jest.mock('../Card', () => ({
  Card: 'CardMock',
}));

jest.mock('../Button', () => ({
  Button: 'ButtonMock',
}));

jest.mock('../Input', () => ({
  Input: 'InputMock',
}));

// Mock react-native components
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');
  return {
    ...RN,
    TouchableOpacity: 'TouchableOpacityMock',
    View: 'ViewMock',
  };
});

// Mock theme
jest.mock('@/constants/theme', () => ({
  COLORS: {
    primary: {
      green: '#00C853',
      blue: '#2979FF',
      red: '#FF5252',
      yellow: '#FFD600',
      purple: '#9C27B0',
      orange: '#FF9800',
      teal: '#00BCD4',
    },
    ui: {
      background: 'rgba(28, 14, 45, 0.95)',
      card: 'rgba(38, 20, 60, 0.85)',
      text: '#FFFFFF',
      textSecondary: 'rgba(255, 255, 255, 0.7)',
      border: 'rgba(82, 67, 194, 0.3)',
      disabled: 'rgba(82, 67, 194, 0.2)',
      accent: '#4169E1',
      highlight: 'rgba(65, 105, 225, 0.15)',
    },
    status: {
      success: '#00E676',
      error: '#FF5252',
      warning: '#FFD600',
      info: '#4169E1',
    },
    gradient: {
      start: 'rgba(65, 105, 225, 0.2)',
      middle: 'rgba(147, 112, 219, 0.15)',
      end: 'rgba(28, 14, 45, 0.9)',
    },
  },
  FONTS: {
    sizes: {
      xs: 12,
      sm: 14,
      md: 16,
      lg: 22,
      xl: 28,
      xxl: 36,
      xxxl: 44,
    },
    weights: {
      regular: '400',
      medium: '500',
      semibold: '600',
      bold: '700',
    },
    families: {
      heading: 'System',
      body: 'System',
    }
  },
  SPACING: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 40,
  },
  BORDER_RADIUS: {
    sm: 4,
    md: 8,
    lg: 16,
    xl: 24,
    circle: 999,
  },
  TIME_PERIODS: {
    MORNING: {
      label: 'Morning',
      range: { start: 5, end: 11 },
      greeting: 'Good morning',
      icon: '🌅'
    },
    AFTERNOON: {
      label: 'Afternoon',
      range: { start: 12, end: 16 },
      greeting: 'Good afternoon',
      icon: '☀️'
    },
    EVENING: {
      label: 'Evening',
      range: { start: 17, end: 4 },
      greeting: 'Good evening',
      icon: '🌙'
    }
  },
  default: {
    COLORS: {},
    FONTS: {},
    SPACING: {},
    BORDER_RADIUS: {},
    TIME_PERIODS: {}
  }
}));

describe('DailyChallenge Component', () => {
  // Reset all mocks before each test
  beforeEach(() => {
    jest.clearAllMocks();
  });

  // Set up mock challenge data
  const mockDailyChallenge = {
    id: 'challenge-1',
    title: 'Daily Gratitude',
    description: 'Write down three things you are grateful for today',
    type: 'gratitude',
    points: 50,
    challenge_status: 'not_started',
  };

  const mockUserChallenges = [
    {
      id: 'user-challenge-1',
      challenge_id: 'challenge-2',
      status: 'completed',
      completed_at: '2023-06-01T12:00:00Z',
      created_at: '2023-06-01T10:00:00Z',
    },
  ];

  const mockUserStats = {
    current_streak: 7,
    longest_streak: 10,
    total_points: 350,
    total_entries: 15,
    last_check_in: '2023-06-02T10:00:00Z',
    level: 3,
  };

  test('renders correctly with daily challenge', () => {
    // Setup mock implementation
    require('@/contexts/ChallengesContext').useChallenges.mockReturnValue({
      dailyChallenge: mockDailyChallenge,
      userChallenges: mockUserChallenges,
      userStats: mockUserStats,
      completeChallenge: jest.fn(),
      refreshDailyChallenge: jest.fn(),
    });

    const { toJSON } = render(<DailyChallenge />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('renders correctly with no daily challenge', () => {
    // Setup mock implementation with null daily challenge
    require('@/contexts/ChallengesContext').useChallenges.mockReturnValue({
      dailyChallenge: null,
      userChallenges: mockUserChallenges,
      userStats: mockUserStats,
      completeChallenge: jest.fn(),
      refreshDailyChallenge: jest.fn(),
    });

    const { toJSON } = render(<DailyChallenge />);
    expect(toJSON()).toMatchSnapshot('no-challenge');
  });

  test('renders correctly when daily limit is reached', () => {
    // Setup mock implementation with completed challenges for today
    const todayDateString = new Date().toISOString(); // Today's date in ISO format
    const completedChallenges = [
      {
        id: 'user-challenge-1',
        challenge_id: 'challenge-1',
        status: 'completed',
        completed_at: todayDateString,
        created_at: todayDateString,
      },
      {
        id: 'user-challenge-2',
        challenge_id: 'challenge-2',
        status: 'completed',
        completed_at: todayDateString,
        created_at: todayDateString,
      },
    ];

    require('@/contexts/ChallengesContext').useChallenges.mockReturnValue({
      dailyChallenge: mockDailyChallenge,
      userChallenges: completedChallenges,
      userStats: mockUserStats,
      completeChallenge: jest.fn(),
      refreshDailyChallenge: jest.fn(),
    });

    const { toJSON } = render(<DailyChallenge />);
    expect(toJSON()).toMatchSnapshot('daily-limit-reached');
  });

  test('renders correctly with daily challenge and provides context functions', () => {
    // Setup mock functions
    const mockRefreshDailyChallenge = jest.fn();
    const mockCompleteChallenge = jest.fn();
    
    require('@/contexts/ChallengesContext').useChallenges.mockReturnValue({
      dailyChallenge: mockDailyChallenge,
      userChallenges: mockUserChallenges,
      userStats: mockUserStats,
      completeChallenge: mockCompleteChallenge,
      refreshDailyChallenge: mockRefreshDailyChallenge,
    });

    // Render component
    const { toJSON } = render(<DailyChallenge />);
    
    // Verify the component rendered correctly with the right context
    expect(toJSON()).toBeTruthy();
    
    // Check that the context functions were provided correctly
    expect(mockRefreshDailyChallenge).toBeDefined();
    expect(mockCompleteChallenge).toBeDefined();
  });
}); 