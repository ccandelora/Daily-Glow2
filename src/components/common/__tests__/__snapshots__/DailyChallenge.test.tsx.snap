// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyChallenge Component renders correctly when daily limit is reached: daily-limit-reached 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <CardMock
    style={
      {
        "backgroundColor": "rgba(38, 20, 60, 0.85)",
        "padding": 24,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "padding": 32,
          "paddingVertical": 48,
        }
      }
    >
      <FontAwesome6Mock
        color="#00C853"
        name="circle-check"
        size={48}
        style={
          {
            "marginBottom": 24,
          }
        }
      />
      <TypographyMock
        style={
          {
            "color": "#00C853",
            "fontSize": 28,
            "fontWeight": "600",
            "marginBottom": 16,
            "textAlign": "center",
          }
        }
        variant="h2"
      >
        Daily Challenges Complete!
      </TypographyMock>
      <TypographyMock
        style={
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 16,
            "lineHeight": 24,
            "marginBottom": 48,
            "textAlign": "center",
          }
        }
      >
        Great job! You've completed your daily challenges. Come back tomorrow for new challenges and more opportunities to earn points.
      </TypographyMock>
      <View
        style={
          {
            "alignItems": "center",
            "marginBottom": 48,
          }
        }
      >
        <TypographyMock
          style={
            {
              "color": "#00C853",
              "fontSize": 72,
              "fontWeight": "700",
              "lineHeight": 80,
              "marginBottom": 4,
              "textAlign": "center",
            }
          }
          variant="h1"
        >
          350
        </TypographyMock>
        <TypographyMock
          style={
            {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontSize": 22,
              "textAlign": "center",
            }
          }
        >
          Total Points
        </TypographyMock>
      </View>
      <ButtonMock
        onPress={[Function]}
        style={
          {
            "marginTop": 16,
            "paddingVertical": 16,
            "width": "100%",
          }
        }
        title="View Your Achievements"
        variant="secondary"
      />
    </View>
  </CardMock>
</View>
`;

exports[`DailyChallenge Component renders correctly with daily challenge 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <CardMock
    style={
      {
        "backgroundColor": "rgba(38, 20, 60, 0.85)",
        "padding": 24,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "gap": 16,
          "justifyContent": "space-between",
          "marginBottom": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "gap": 8,
            "minWidth": 150,
          }
        }
      >
        <FontAwesome6Mock
          color="#00C853"
          name="face-smile"
          size={24}
        />
        <TypographyMock
          style={
            {
              "color": "#FFFFFF",
              "flexShrink": 1,
              "marginLeft": 4,
              "textShadowColor": "#4169E1",
              "textShadowOffset": {
                "height": 0,
                "width": 0,
              },
              "textShadowRadius": 8,
            }
          }
          variant="h3"
        >
          Daily Challenge
        </TypographyMock>
      </View>
      <View
        style={
          {
            "alignItems": "flex-end",
            "flexShrink": 0,
            "minWidth": 100,
          }
        }
      >
        <TypographyMock
          color="rgba(255, 255, 255, 0.7)"
          variant="caption"
        >
          Level 
          3
        </TypographyMock>
        <TypographyMock
          color="#00C853"
          style={
            {
              "color": "#00C853",
              "flexShrink": 0,
              "flexWrap": "wrap",
              "fontWeight": "700",
              "marginTop": 4,
              "minWidth": 100,
              "textAlign": "right",
              "textShadowColor": "#00C853",
              "textShadowOffset": {
                "height": 0,
                "width": 0,
              },
              "textShadowRadius": 4,
            }
          }
          variant="h3"
        >
          350
           pts
        </TypographyMock>
      </View>
    </View>
    <TypographyMock
      style={
        {
          "color": "#FFFFFF",
          "marginBottom": 8,
          "textShadowColor": "#4169E1",
          "textShadowOffset": {
            "height": 0,
            "width": 0,
          },
          "textShadowRadius": 6,
        }
      }
      variant="h3"
    >
      Daily Gratitude
    </TypographyMock>
    <TypographyMock
      style={
        {
          "color": "rgba(255, 255, 255, 0.7)",
          "marginBottom": 24,
        }
      }
      variant="body"
    >
      Write down three things you are grateful for today
    </TypographyMock>
    <InputMock
      maxLength={500}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Share what you're grateful for today..."
      style={
        {
          "backgroundColor": "rgba(28, 14, 45, 0.3)",
          "borderColor": "rgba(82, 67, 194, 0.3)",
          "height": 120,
          "marginBottom": 24,
          "textAlignVertical": "top",
        }
      }
      value=""
    />
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 4,
            "justifyContent": "center",
          }
        }
      >
        <FontAwesome6Mock
          color="#00C853"
          name="star"
          size={16}
          style={
            {
              "marginRight": 8,
            }
          }
        />
        <TypographyMock
          style={
            {
              "color": "#00C853",
            }
          }
          variant="caption"
        >
          +
          50
           points
        </TypographyMock>
      </View>
    </View>
    <ButtonMock
      disabled={false}
      onPress={[Function]}
      style={
        {
          "backgroundColor": "rgba(65, 105, 225, 0.15)",
          "borderColor": "#4169E1",
          "borderWidth": 1,
          "elevation": 4,
          "marginTop": 16,
          "shadowColor": "#4169E1",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 8,
          "width": "100%",
        }
      }
      textStyle={
        {
          "color": "#00C853",
        }
      }
      title="Complete Challenge"
      variant="secondary"
    />
    <View
      style={
        {
          "alignItems": "center",
          "borderTopColor": "rgba(82, 67, 194, 0.3)",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "marginTop": 16,
          "paddingTop": 16,
        }
      }
    >
      <FontAwesome6Mock
        color="#FF5252"
        name="flame"
        size={16}
        style={
          {
            "marginRight": 8,
          }
        }
      />
      <TypographyMock
        style={
          {
            "color": "#FF5252",
            "fontWeight": "700",
          }
        }
        variant="caption"
      >
        7
         day streak
      </TypographyMock>
    </View>
  </CardMock>
</View>
`;

exports[`DailyChallenge Component renders correctly with no daily challenge: no-challenge 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <CardMock
    style={
      {
        "backgroundColor": "rgba(38, 20, 60, 0.85)",
        "padding": 24,
        "width": "100%",
      }
    }
  >
    <TypographyMock
      style={
        {
          "color": "rgba(255, 255, 255, 0.7)",
          "lineHeight": 24,
          "marginBottom": 32,
          "paddingHorizontal": 24,
          "textAlign": "center",
          "width": "100%",
        }
      }
      variant="body"
    >
      No challenges available right now. Please check back later.
    </TypographyMock>
  </CardMock>
</View>
`;
